modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/LayoutProvider/handlers/fieldDeletedHandler.es.js

```diff
@@ -96,17 +101,17 @@ export const removeField = (props, pages, fieldName) => {
 					},
 				]);
 
+				const pages = visitor.mapColumns((column) => ({
+					...column,
+					fields: column.fields.filter(
+						(nestedFieldName) => fieldName !== nestedFieldName
+					),
+				}));
+
 				const rows = field.rows
-					? FormSupport.removeEmptyRows(
-							visitor.mapColumns((column) => ({
-								...column,
-								fields: column.fields.filter(
-									(nestedFieldName) =>
-										fieldName !== nestedFieldName
-								),
-							})),
-							0
-					  )
+					? removeEmptyRows
+						? FormSupport.removeEmptyRows(pages, 0)
+						: pages[0].rows
```

Comment:
@carolmariaabb, please avoid use nested ternary operator because it decreases code legibility.